internals/overlord/checkstate/handlers.go

```diff
@@ -42,6 +42,13 @@ func (m *CheckManager) doPerformCheck(task *state.Task, tomb *tombpkg.Tomb) erro
 	ticker := time.NewTicker(config.Period.Value)
 	defer ticker.Stop()
 
+	// Retrieve checkData and its channels
+	m.checksLock.Lock()
+	m.ensureCheck(config.Name)
+	checkData := m.checks[details.Name]
+	refresh := checkData.refresh
```

Comment:
`ensureCheck` returns the data, so how about this:

```go
data := m.ensureCheck(config.Name)
refresh := data.refresh
```

Or even this (but up to you which is clearer):

```go
refresh := m.ensureCheck(config.Name).refresh
```